codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -554,33 +583,37 @@ final void weight(short weight) {
          */
         final void takeChild(DefaultStream child, boolean exclusive, List<ParentChangedEvent> events) {
             DefaultStream oldParent = child.parent();
-            events.add(new ParentChangedEvent(child, oldParent));
-            notifyParentChanging(child, this);
-            child.parent = this;
 
-            if (exclusive && !children.isEmpty()) {
-                // If it was requested that this child be the exclusive dependency of this node,
-                // move any previous children to the child node, becoming grand children of this node.
-                for (DefaultStream grandchild : removeAllChildren().values()) {
-                    child.takeChild(grandchild, false, events);
+            if (oldParent != this) {
+                events.add(new ParentChangedEvent(child, oldParent));
+                notifyParentChanging(child, this);
+                child.parent = this;
+                // We need the removal operation to happen first so the prioritizableForTree for the old parent to root
+                // path is updated with the correct child.prioritizableForTree() value.
+                if (oldParent != null && oldParent.children.remove(child.id()) != null) {
```

Comment:
I don't think so.  When a stream is created the connection stream does a `takeChild` on that stream....and its parent will be null.  After this it would be an error condition though...were you thinking of having the connection stream override this method and have the mainstream case not have to worry about the null check?